src/platform/packages/shared/kbn-unified-metrics-grid/src/components/metrics_grid_toolbar_wraper.tsx

```diff
@@ -0,0 +1,120 @@
+import type { SerializedStyles } from '@emotion/serialize';
+import type { MetricField } from '@kbn/metrics-experience-plugin/common/types';
+import type { ChartSectionProps } from '@kbn/unified-histogram/types';
+import { useMetricsGridState } from '../hooks';
+import { RightSideActions } from './toolbar/right_side_actions/search_input';
+import { useToolbarActions } from './toolbar/hooks/use_toolbar_actions';
+
+const getFullScreenStyles = (euiTheme: EuiThemeComputed): CSSObject => {
+  return {
+    position: 'fixed',
+    top: 0,
+    left: 0,
+    width: '100vw',
+    height: '100vh',
+    background: euiTheme.colors.backgroundBasePlain,
+    display: 'flex',
+    flexDirection: 'column',
+    minHeight: 0,
+    zIndex: euiTheme.levels.modal,
+    overscrollBehavior: 'contain',
+  };
+};
+
+interface FullScreenProps {
+  isFullscreen: boolean;
+  dataTestSubj?: string;
+}
+
+const FullScreenWrapper = ({
+  isFullscreen,
+  dataTestSubj,
+  children,
+}: React.PropsWithChildren<FullScreenProps>) => {
+  const { euiTheme } = useEuiTheme();
+
+  const styles = useMemo(() => getFullScreenStyles(euiTheme), [euiTheme]);
+
+  if (!isFullscreen) return <>{children}</>;
+
+  return (
+    <EuiOverlayMask headerZindexLocation="above">
+      <EuiFocusTrap>
+        <div css={styles} data-test-subj={`${dataTestSubj}FullScreenWrapper`}>
+          {children}
+        </div>
+      </EuiFocusTrap>
+    </EuiOverlayMask>
+  );
+};
+
+export interface MetricsGridToolbarWrapperProps
+  extends Pick<ChartSectionProps, 'requestParams' | 'renderToggleActions'> {
+  indexPattern: string;
+  chartToolbarCss?: SerializedStyles;
+  setDebouncedSearchTerm: (value: string) => void;
+  fields: MetricField[];
+  children?: React.ReactNode;
+}
+
+export const MetricsGridToolbarWrapper = ({
+  indexPattern,
+  renderToggleActions,
+  chartToolbarCss,
+  requestParams,
+  setDebouncedSearchTerm,
+  fields,
+  children,
+}: MetricsGridToolbarWrapperProps) => {
+  const { leftSideActions } = useToolbarActions({
+    fields,
+    indexPattern,
+    renderToggleActions,
+    setDebouncedSearchTerm,
+    requestParams,
+  });
+  const { searchTerm, onSearchTermChange, isFullscreen } = useMetricsGridState();
+
+  return (
+    <FullScreenWrapper isFullscreen={isFullscreen} dataTestSubj="metricsExperienceGrid">
+      <ChartSectionTemplate
+        id="metricsExperienceGridPanel"
+        toolbarCss={chartToolbarCss}
+        toolbar={{
+          leftSide: leftSideActions,
+          additionalControls: {
+            prependRight: (
+              <RightSideActions
+                key="rightSideActions"
+                searchTerm={searchTerm}
+                onSearchTermChange={onSearchTermChange}
+                fields={fields}
+                indexPattern={indexPattern}
+                renderToggleActions={renderToggleActions}
+                requestParams={requestParams}
+                setDebouncedSearchTerm={setDebouncedSearchTerm}
```

Comment:
Why do we need `onSearchTermChange` and `setDebouncedSearchTerm`. Don't they do the same thing?